File: mycollab-ui/src/main/scala/com/esofthead/mycollab/vaadin/events/MassItemActionHandler.scala
package com.esofthead.mycollab.vaadin.events

import com.esofthead.mycollab.reporting.ReportExportType
import com.vaadin.server.StreamResource

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
trait MassItemActionHandler {
  def onSelect(id: String)

  def buildStreamResource(exportType: ReportExportType): StreamResource
}

object ViewItemAction {
  val MAIL_ACTION: String = "mail"
  val DELETE_ACTION: String = "delete"
  val MASS_UPDATE_ACTION: String = "massUpdate"
}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/events/MassItemActionHandler.kt
package com.mycollab.vaadin.events

import com.mycollab.reporting.ReportExportType
import com.vaadin.server.StreamResource

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
interface MassItemActionHandler {
    fun onSelect(id: String)

    fun buildStreamResource(exportType: ReportExportType): StreamResource
}

object ViewItemAction {
    @JvmField val MAIL_ACTION = "mail"
    @JvmField val DELETE_ACTION = "delete"
    @JvmField val MASS_UPDATE_ACTION = "massUpdate"
}